### DISCUSSION TITLE
Feature enhancement for Osmand+ BLE Tire Pressure Monitoring (FOBO)

### DISCUSSION THREAD

**Discussion by svigr1** [osmandapp/OsmAnd#18310] [Ideas]
To whom it may concern,

I'm an owner of 5 motorcycles and have OsmAnd+ Pro. I'm requesting to see if we can get support for 3rd party Bluetooth/ BLE for tire pressure monitoring systems and add a widget to be able to see tire pressure for both Front and Rear Tire for the solution below;

FOBO Bike 2 tire pressure monitoring system (Black) – external monitor, bike tire, temperature sensor, wireless, for smart bike, motorcycle, ebike & bicycle  https://a.co/d/iEff9il

The Fobo Bike 2 is a tire pressure monitoring system (TPMS) that uses Bluetooth 5.0 technology to monitor tire pressure in real time. It can detect slow or fast leaks and alert users to anomalies. The Fobo Bike 2 works with iOS 9.3 and Android 5.0 or later. This can provide safety to anyone with two-wheel vehicles and help eliminate crashes due to low tire pressure.   Plus, it gives you an active tire pressure when riding to know that the tire is also up to temperature. 

Please let me know if this is feasible.

V/R

Stan

**Comment by mikehgentry** [comment]
Probably a duplicate of #18208 assuming they all use the same protocol.

**Reply by svigr1** [reply]
I figure I would add this


API info on Fobo 2

https://github.com/karma4u101/FoBo

**Reply by NH-RedAnt** [reply]
I have recently been selecting a personal motorcycle BLE TPMS  sensor like the Fobo2. From what I have learned there are many different formats used by a variety of vendors. I  used the "BLE Scanner" app to see packets from the earlier Fobo exrernal sensor.

In reviewing the API link provided above, I don't believe this FoBo API has anything to do with the tire pressure sensor hardware. Seems to be only a name collision of very different things. 

There does appear to be one external BLE sensor whos data encoding has been decoded. Unfortunately it sems to be sold under many product names.

I ordered one of the newer Bluetooth 5 variants to test. If it's data can be decoded, I will summarize the manufacturer tjat I purchased and the known coding link, as well as my own findings.

My goal is to plot my tires pressures plotted over time. So I  wouldn't mind looking into getting this data into the osmand plugin.  Any examples that describe what is required by osmand would be greatfuly accepted.

**Comment by svigr1** [comment]
Thank you for doing so. I'm eagerly waiting to hear of your results. 

V/R
Stan

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Aug 1, 2024 at 5:59 PM, Robert E ***@***.***> wrote:   


I have recently been selecting a personal motorcycle BLE TPMS sensor like the Fobo2. From what I have learned there are many different formats used by a variety of vendors. I used the "BLE Scanner" app to see packets from the earlier Fobo exrernal sensor.

In reviewing the API link provided above, I don't believe this FoBo API has anything to do with the tire pressure sensor hardware. Seems to be only a name collision of very different things.

There does appear to be one external BLE sensor whos data encoding has been decoded. Unfortunately it sems to be sold under many product names.

I ordered one of the newer Bluetooth 5 variants to test. If it's data can be decoded, I will summarize the manufacturer tjat I purchased and the known coding link, as well as my own findings.

My goal is to plot my tires pressures plotted over time. So I wouldn't mind looking into getting this data into the osmand plugin. Any examples that describe what is required by osmand would be greatfuly accepted.
